### ISSUE TITLE
Failed to execute 'postMessage' on 'Window'

### ISSUE THREAD

**Issue by Nicolab** [goatslacker/alt-devtool#5]
Hello,

there is an error in the console log.

```
Exception: Failed to execute 'postMessage' on 'Window': An object could not be cloned. 
chrome-extension://hlphpmkmonogpcpobpgjoidchbgbplio/dist/altif.js:122 
```

```
code: 25
message: "Failed to execute 'postMessage' on 'Window': An object could not be cloned.
"name: "DataCloneError
"stack: 
"Error: Failed to execute 'postMessage' on 'Window': An object could not be cloned.
↵    at Error (native)
↵    at post (chrome-extension://hlphpmkmonogpcpobpgjoidchbgbplio/dist/altif.js:122:10
```

But the extension work well :)

---

Chromium Version 41.0.2272.76 Ubuntu 15.04 (64-bit)

**Comment by tomaash** [comment]
I'm having this issue as well. This happens when you post more complicated objects to dispatcher. For example an request configuration:

``` js
StatusActions.failed({config: err.config, action: context.actionDetails});
```

It goes through dispatcher without problems, but the extension needs to use postMessage, and some objects are too much for it's serialization engine. When I put the data through JSON.stringify, it does not cause this error. Solution might be to use JSON.stringify when postMessaging an object of type Object to  the extension. @goatslacker what do you think?

**Comment by pstoica** [comment]
I just had this occur for bound functions. I think stringifying before postMessage would be the safest bet. (I'm passing functions to deal with some weird stuff, it'll hopefully go away soon.)

**Comment by goatslacker** [comment]
https://github.com/goatslacker/alt-devtool/blob/master/scripts/utils/post.js#L4 It's here.

Some things might need to change since I already stringify some payloads, I think? We don't want to double stringify

**Comment by renfredxh** [comment]
I was having this issue as well and the errors it was throwing caused parts of my app to break so I tracked it down. The issue is with `post` but only when it's being called for dispatches [at this spot](https://github.com/goatslacker/alt-devtool/blob/3db1bda1e04dfdd3b67265a032ecbdb03560a8ae/scripts/utils/registerAlt.js#L126-L131).

The exception is caused by the browser's implementation of the [structured cloning algorithm](https://developer.mozilla.org/en-US/docs/Web/Guide/API/DOM/The_structured_clone_algorithm), which `Window.postMessage` uses to clone objects. The implementation of the cloning algorithm causes an error to be thrown when attempting to clone error, function, and DOM node objects. 

In my case, I was passing a fetch response object from axios through the payload which contains function and error objects, thus causing the clone method inside `Window.postMessage` to throw the error. 

> Some things might need to change since I already stringify some payloads, I think? We don't want to double stringify

You're only calling stringify in [one other place](https://github.com/goatslacker/alt-devtool/blob/c25c63cd8a953e8387b08b188843d5e9e9ab8838/scripts/utils/getStoreData.js#L4), which is for data related to Stores.  This is why the error only shows up for dispatches, since the store's data is already safely stringifyed. I made a fix (see PR #11) that calls `JSON.stringify` on the dispatchers' payloads as well. After that everything runs perfectly for me.
